nilearn/decoding/decoder.py

```diff
@@ -583,6 +617,33 @@ def fit(self, X, y, groups=None):
                 if not self.n_outputs_ > 1:
                     self.dummy_output_ = self.dummy_output_[0, :][np.newaxis, :]
 
+    def score(self, X, y, *args):
+        """Compute the prediction score using the scoring
+        metric defined by the scoring attribute.
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix}, shape = (n_samples, n_features)
+            Samples.
+
+        y : array-like
+            Target values.
+
+        args : Optional arguments that can be passed to
+            scoring metrics. Example: sample_weight.
+
+        Returns
+        -------
+        score : float
+            Prediction score.
+
+        """
+        if self.scoring is None:
+            raise ValueError("Unable to compute score. "
+                             "No scoring metric was provided "
+                             "to Decoder object.")
```

Comment:
But in that case, why not relay on a default metric depending on the nature of the classifier ? e.g. accuracy for a classifier, r2 for a regressor.

Reply:
I feel I'm missing something, sorry...

Reply:
Suggestions (that might be naive as I'm not used to nilearn/sklearn) regarding the way to get the scorer:

- maybe we could use the `get_scorer` method instead of storing a list of available scorers on nilearn side?
- then we could have a `default_scorer` with value `"accuracy"` or `"r2"` depending on the `self.is_classification` flag

Reply:
Thanks for the reviews @bthirion @RaphaelMeudec!

You're right, using `get_scorer` is much better :+1:. For some reason I thought that the string values of `decoder.scoring` wouldn't be valid sklearn scorer keys, but they totally are... 

>I feel I'm missing something, sorry...

I don't think so. I hesitated between relying on a default scoring metric and raising an error, and went for the last option to make sure the user doesn't get unexpected results. I agree this is over-restrictive and changed the behaviour to rely on a default metric depending on the nature of the classifier like you both suggested.